lib/cluster.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+/**
+ * In cluster masters, extends Registry class with a `clusterMetrics` method,
+ * which returns aggregated metrics for all workers.
+ *
+ * In cluster workers, listens for and responds to requests for metrics by the
+ * cluster master.
+ */
+
+const cluster = require('cluster');
+const Registry = require('./registry');
+const util = require('./util');
+
+const GET_METRICS_REQ = 'prom-client:getMetricsReq';
+const GET_METRICS_RES = 'prom-client:getMetricsRes';
+
+if (cluster.isMaster) {
+	let requestCtr = 0; // Concurrency control
+	const requests = new Map();
+
+	// Listener for worker responses to requests for local metrics
+	cluster.on('message', (worker, message) => {
+		if (arguments.length === 2) {
+			// pre-Node.js v6.0
+			message = worker;
+			worker = undefined;
+		}
+
+		if (message.type === GET_METRICS_RES) {
+			const request = requests.get(message.requestId);
+			request.responses.push(message.metrics);
+			request.pending--;
+
+			if (request.pending === 0) {
+				// finalize
+				requests.delete(message.requestId);
+				const aggregatedRegistry = Registry.aggregate(request.responses);
+				const promString = aggregatedRegistry.metrics();
+				request.callback(null, promString);
+			}
+		}
+	});
+
+	/**
+	 * Returns a new function that applies the `aggregatorFn` to the values.
+	 * @param {Function} aggregatorFn function to apply to values.
+	 * @return {Function} aggregator function
+	 */
+	const AggregatorFactory = function(aggregatorFn) {
+		return metrics => {
+			if (metrics.length === 0) return;
+			const result = {
+				help: metrics[0].help,
+				name: metrics[0].name,
+				type: metrics[0].type,
+				values: []
+			};
+			const byLabels = new Map();
+			metrics.forEach(metric => {
+				metric.values.forEach(value => {
+					const key = util.hashObject(value.labels);
+					if (byLabels.has(key)) {
+						byLabels.get(key).push(value);
+					} else {
+						byLabels.set(key, [value]);
+					}
+				});
+			});
+			byLabels.forEach(values => {
```

Comment:
He's using a Map, I don't thing you can call .map directly, 
he could convert it to an Array using Array.from an go from there.